### ISSUE TITLE
poetry 1.8.2 requires pkginfo 1.10.0

### ISSUE THREAD

**Issue by skycaptain** [Homebrew/homebrew-core#167916]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 4.2.16-60-g48db9ef
ORIGIN: https://github.com/Homebrew/brew
HEAD: 48db9ef662580a163c5f153cfa1819aa294c422d
Last commit: 2 hours ago
Core tap HEAD: 0f61f2950ecc464a9b2cf39b59fdd57b6423b562
Core tap last commit: 6 hours ago
Core tap JSON: 03 Apr 16:53 UTC
Core cask tap JSON: 03 Apr 16:53 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit arm_palma
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false

$ brew doctor
Your system is ready to brew.
```

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I tried to add a package using poetry. I ran into this issue: https://github.com/python-poetry/poetry/issues/9191. This is due Homebrew shipping `pkginfo==1.9.6` instead of `pkginfo==1.10.0`.

Sidenote: Installing poetry via pipx works fine.

### What happened (include all command output)?

I encountered the error mentioned in https://github.com/python-poetry/poetry/issues/9191

### What did you expect to happen?

The poetry command to work.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
See https://github.com/python-poetry/poetry/issues/9191.

Basically:

$ brew install poetry
$ poetry add -e --group=dev ./sdk
```

**Comment by ZhongRuoyu** [comment]
The fix #167917 is in. Please do `brew upgrade poetry`.
